### PR TITLE
Add JobId to X-Pelican-JobId header

### PR DESCRIPTION
**Pull request by joereuss12**
For uploads and downloads we now can see an X-Pelican-JobId header that records the GlobalJobId from a condor classad. I also refactored the GetProjectName function to be just a SearchJobAd function to work with the two different classads we look for.

### CODE REVIEW

**Comment by joereuss12** [comment]
@bbockelm Did you still want to review this or should I give it to someone else?

File: client/main.go
Comment (haoming29) [inline client/main.go:797]: Can you leave a comment here that briefly explain what `SearchJobAd` does?

File: client/main.go
Comment (haoming29) [inline client/main.go:799]: This comment seems to be outdated

Comment (haoming29) [reply client/main.go:799]: Same as the one below (line 811): or you might want to move it to line 821?
> // https://stackoverflow.com/questions/28574609/how-to-apply-regexp-to-content-in-file-go

**Review by haoming29 — APPROVED** [review]
Pre-approved once the minor issues are addressed

File: client/main.go
Comment (bbockelm) [inline client/main.go:799]: Do not export outside the module.

File: client/main.go
Comment (bbockelm) [inline client/main.go:49]: Do not export values.

**Review by bbockelm — COMMENTED** [review]
Please clean up some exports here.

**Comment by joereuss12** [comment]
@haoming29 I know it says it is good to merge but I addressed Brian's requests for changes so could you just give that a glance over before this is merged?

**Comment by haoming29** [comment]
> @haoming29 I know it says it is good to merge but I addressed Brian's requests for changes so could you just give that a glance over before this is merged?

Did a quick peek and the changes look good to me, with one question left

**Pull request merged**
